spec/models/shf_application_spec.rb

```diff
@@ -172,14 +172,9 @@
       expect(uploaded_file.destroyed?).to be_truthy
     end
 
-    it "destroys associated company if it has no remaining applications" do
```

Comment:
Why did you remove this?  Is it because we need to have the explicit discussion with the client about this?  (I'm assuming so, but just wanted to be sure.)

Reply:
It's not really due to the reason you mention (although we'd probably want to do this here, anyway, for that reason).

My _intent_ for this PR was to not change any unrelated logic.

However - _if_ we retained the deletion of the company when the app is deleted - than we will have to adopt another approach altogether (see story I created for this - https://www.pivotaltracker.com/story/show/154296716).  So, if we did this, it is a larger story than would fit into to this story.

(the prior way of destroying the company - when the app is detroyed - involved some rather tortuous logic between the two models - which is no longer viable with the change to `has_and_belongs_to_many` association.